src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java

```diff
@@ -0,0 +1,107 @@
+package org.folio.circulation.resources;
+
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.LoanRepository;
+import org.folio.circulation.domain.notice.schedule.DueDateScheduledNoticeService;
+import org.folio.circulation.domain.representations.ChangeDueDateRequest;
+import org.folio.circulation.domain.validation.ItemStatusValidator;
+import org.folio.circulation.domain.validation.LoanValidator;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.NoContentResult;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.server.WebContext;
+import org.joda.time.DateTime;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class ChangeDueDateResource extends Resource {
+  public static final String DUE_DATE = "dueDate";
```

Comment:
Can we use the one from `ChangeDueDateRequest`?

Reply:
Are you using the word 'request' here to refer to the ILS concept of request, or the HTTP concept? 

It seems like the like the convention of 'resource' has been adopted for the later case, maybe to avoid any confusion.  As far as I can tell this class does not pertain to the ILS concept of a request.

There is already a org.folio.circulation.domain.representations.ChangeDueDateRequest, the name collision would make for some ugliness (in the signature of `processChangeDueDate` for instance.)

Reply:
Sorry @jeremythuff. I meant reusing existing constant from `org.folio.circulation.domain.representations.ChangeDueDateRequest#DUE_DATE` instead of introducing a new one in this class.

Reply:
Resource was my naming convention, it likely turned out to be a pretty poor choice and at some point I intend to rename / restructure these entry points into the system

Reply:
Oops, I thought you wanted to refactor the class :/ That makes more sense.